src/containers/TopicLandingPage.js

```diff
@@ -27,128 +25,67 @@ const _  = {
-      viewportHeight: '100vh'
-    }
+  static defaultProps = {
+    entities: {},
+    selectedTopic: {}
+  }
+  
+  static propTypes = {
+    // from redux store
+    entities: PropTypes.object,
+    selectedTopic: PropTypes.object,
+    // react-router `match` object
+    match: PropTypes.object.isRequired
   }
 
-  componentWillMount() {
+  componentDidMount() {
     const { fetchAFullTopic, match } = this.props
     const slug = _.get(match, 'params.slug')
     fetchAFullTopic(slug)
   }
 
-  componentDidMount() {
-    this._updateViewportHeight()
-  }
-
-  componentDidUpdate() {
-    this._updateViewportHeight()
-  }
-
-  componentWillReceiveProps(nextProps) {
-    const { fetchAFullTopic, match } = nextProps
-    const { selectedTopic } = this.props
-    if (_.get(selectedTopic, 'slug') !== _.get(match, 'params.slug')) {
-      fetchAFullTopic(_.get(match, 'params.slug'))
-    }
-  }
-
-  _updateViewportHeight() {
-    if (typeof window === 'object') {
-      /*
-        The height we want is which of the viewport that actually be seen by users.
-        But the common used DOM properties have several problems:
-          `documentElement.clientHeight` -> In iOS, the return value pretends that the URL and tab bars are visible, even if they are not.
-          `window.innerHeight` -> It includes the area covered by scroll bars (user cannot see the area actually)
-        The best way to get the value may be: (window.innerHeight - scroll bar height)
-        Since this page should not have horizontal scroll bar, we can use `window.innerHeight` directly.
-        Reference: https://stackoverflow.com/a/31655549
-      */
-      const heightString = window.innerHeight ? `${window.innerHeight}px` : '100vh'
-      if (this.state.viewportHeight !== heightString) {
-        this.setState({
-          viewportHeight: heightString
-        })
-      }
-    }
-  }
-
   render() {
     const { entities, match, selectedTopic } = this.props
-    const { viewportHeight } = this.state
-    const error = _.get(selectedTopic, 'error', null)
-    if (error !== null) {
+    const error = _.get(selectedTopic, 'error')
+    if (error) {
       return (
-        <div>
-          <SystemError error={error} />
-        </div>
+        <SystemError error={error} />
       )
     }
-
-    if (_.get(selectedTopic, 'slug') !== _.get(match, 'params.slug')) {
-      return null
-    }
-
     const slug = _.get(selectedTopic, 'slug') // {string}
+    if (slug !== _.get(match, 'params.slug')) {
+      console.warn('The `selectedTopic.slug` in redux store does not match the slug in url.') // eslint-disable-line no-console
+    }
 
     const postEntities = _.get(entities, reduxStateFields.postsInEntities, {})
     const topicEntities = _.get(entities, reduxStateFields.topicsInEntities, {})
-    const topic = camelizeKeys(utils.denormalizeTopics(slug, topicEntities, postEntities)[0])
+    const topic = utils.denormalizeTopics(slug, topicEntities, postEntities)[0]
```

Comment:
`const topic = _.get(utils.denormalizeTopics(slug, topicEntities, postEntities), 0, {})` might be safer?